From 10c2006e627b5bda74f9f8aca1a574aa0e0e1eaa Mon Sep 17 00:00:00 2001
From: Kai Krakow <kai@kaishome.de>
Date: Sun, 9 Jul 2023 12:24:40 +0200
Subject: [PATCH] HACK: crucible: Work around kernel memory fragmentation

According to @Zygo:

> This will drastically reduce the number of reflinks bees can handle
> to a single extent, but that size is beyond more than enough for most
> filesystems. Making the buffer smaller may also reduce the size of
> vmallocs which might be aggravating the kernel's memory manager.

Link: https://github.com/Zygo/bees/issues/260#issuecomment-1627586574
Signed-off-by: Kai Krakow <kai@kaishome.de>
---
 include/crucible/fs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/crucible/fs.h b/include/crucible/fs.h
index 6d1780b..5da7da0 100644
--- a/include/crucible/fs.h
+++ b/include/crucible/fs.h
@@ -65,7 +65,7 @@ namespace crucible {
 	};
 
 	struct BtrfsIoctlLogicalInoArgs {
-		BtrfsIoctlLogicalInoArgs(uint64_t logical, size_t buf_size = 16 * 1024 * 1024);
+		BtrfsIoctlLogicalInoArgs(uint64_t logical, size_t buf_size = 64 * 1024);
 
 		uint64_t get_flags() const;
 		void set_flags(uint64_t new_flags);
-- 
2.39.3

